ksql-functional-tests/src/test/resources/query-validation-tests/key-field.json

```diff
@@ -90,30 +90,30 @@
         "CREATE STREAM INPUT (foo INT, bar INT) WITH (kafka_topic='input_topic', value_format='JSON');",
         "CREATE STREAM OUTPUT AS SELECT foo AS ALIASED, bar FROM INPUT PARTITION BY ALIASED;"
       ],
+      "expectedException": {
+        "type": "io.confluent.ksql.util.KsqlStatementException",
+        "message": "Invalid identifier for PARTITION BY clause: ALIASED"
```

Comment:
work adding... `Only columns from the source schema can be referenced in the PARTITION BY clause`?